src/client.rs

```diff
@@ -47,21 +49,134 @@ pub struct Client {
     inner: Arc<ClientRef>,
 }
 
-impl Client {
-    /// Constructs a new `Client`.
-    pub fn new() -> ::Result<Client> {
-        let mut client = try_!(new_hyper_client());
-        client.set_redirect_policy(::hyper::client::RedirectPolicy::FollowNone);
+/// Represent an X509 certificate.
+pub struct Certificate(hyper_native_tls::Certificate);
+
+impl Certificate {
+    /// Create a `Certificate` from a binary DER encoded certificate
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
+    /// # use std::io::Read;
+    /// # fn main() {
+    /// let mut buf = Vec::new();
+    /// let _ = ::std::fs::File::open("my-cert.der").unwrap().read_to_end(&mut buf).unwrap();
+    /// let cert = reqwest::Certificate::from_der(&buf).unwrap();
+    /// # }
+    /// ```
```

Comment:
The Rust community is starting to encourage a different way of writing doc examples, that don't use `unwrap` and stuff. It could be written like this:

```
/// ```
/// # use std::fs::File;
/// # fn cert() -> Result<(), Box<std::error::Error>> {
/// let mut buf = Vec::new();
/// File::open("my_cert.der")?
///     .read_to_end(&mut buf)?;
/// let cert = reqwest::Certificate::from_der(&buf)?;
/// # drop(cert);
/// # Ok(())
/// # }
/// ```
```